examples/06-custom-effects/README.md

```diff
@@ -1,24 +1,74 @@
+![](./header.png)
+
 # 06 Custom Effects
 
-We haven't quite got round to documenting this example yet. If you know a little
-bit about Lustre or Elm and want to help out, we'd love to have your help! Please
-[open an issue](https://github.com/lustre-labs/lustre/issues/new) if you have any
-ideas or reach out to @hayleigh.dev on the [Gleam discord](https://discord.gg/Fm8Pwmy).
+In the last example, we showed how to use effects provided by `lustre_http`. In
+this example we'll see how to create effects of our own using Lustre's
+[`effect.from`](https://hexdocs.pm/lustre/4.0.0-rc1/lustre/effect.html#from)
+method.
+
+Since we use Lustre effects to communicate with _external_ systems (like the
+browser or the Erlang VM) you'll find creating custom effects usually involves
+Gleam's [external
+functions](https://tour.gleam.run/everything/#advanced-features-externals). So
+be sure to read up on that!
+
+> Gleam externals are part of its "foreign function interface", which is why
+> you'll typically see files with `ffi` in the name - like
+> [`app.ffi.mjs`](./src/app.ffi.mjs).
+
+## Accessing Browser Storage
+
+In this example, the external system we want to interact with is the browser's
+local storage. This way, we can write a message into the text input and it will
```

Comment:
Let's link out to the `localStorage` docs as a courtesy: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage